### PR TITLE
Adds a utility class for working with the PicoContainer in tests.

### PR DESCRIPTION
**Pull request by nkibler**
Progress towards #1407.

This is a test utility that replaces registered components in the MutablePicoContainer with mocks and keeps track of the original instances, which have to be replaced when the test is tearing down.

### CODE REVIEW

File: google-cloud-tools-plugin/testSrc/com/intellij/appengine/PicoContainerTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        (MutablePicoContainer) ApplicationManager.getApplication().getPicoContainer();
+    Object originalInstance = applicationContainer.getComponentInstanceOfType(clazz);
+    applicationContainer.unregisterComponent(clazz.getName());
+    applicationContainer.registerComponentInstance(clazz.getName(), newInstance);
+    return originalInstance;
+  }
+
+  /** Represents a bound component in the {@link MutablePicoContainer}. */
+  @AutoValue
+  abstract static class Component {
```
Comment (nkibler) [inline google-cloud-tools-plugin/testSrc/com/intellij/appengine/PicoContainerTestUtil.java]: I thought the same thing, but figured we could benefit from it elsewhere potentially in the future. It's also only marked as a test dep so it won't affect the plugin's built size.

File: google-cloud-tools-plugin/testSrc/com/intellij/appengine/PicoContainerTestUtil.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intellij.appengine;
+
+import com.google.auto.value.AutoValue;
+
+import com.intellij.openapi.application.ApplicationManager;
+
```
Comment (patflynn) [inline google-cloud-tools-plugin/testSrc/com/intellij/appengine/PicoContainerTestUtil.java]: these imports seem oddly spaced.

Comment (nkibler) [inline google-cloud-tools-plugin/testSrc/com/intellij/appengine/PicoContainerTestUtil.java]: This is how the IntelliJ import optimizer is formatting it, but I agree, it shouldn't be like this. In fact, Google recently revised their style rules to say all imports should be in the same block: https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing

Should I use this approach (static in one block, others in another block) on all new files?

**Comment by patflynn** [comment]
are you running google java format?

**Comment by nkibler** [comment]
Yep, but I think the project is configured to split up the imports:
screenshot from 2017-07-19 12 58 46

**Pull request merged**